nuxeo-features/nuxeo-platform-pdf-utils/src/main/java/org/nuxeo/ecm/platform/pdf/PDFInfo.java

```diff
@@ -0,0 +1,477 @@
+
+    private AccessPermission permissions;
+
+    private LinkedHashMap<String, String> cachedMap;
+
+    /**
+     * Constructor with a Blob.
+     *
+     * @param inBlob Input blob.
+     */
+    public PDFInfo(Blob inBlob) {
+        this(inBlob, null);
+    }
+
+    /**
+     * Constructor for Blob + encrypted PDF.
+     *
+     * @param inBlob Input blob.
+     * @param inPassword If the PDF is encrypted.
+     */
+    public PDFInfo(Blob inBlob, String inPassword) {
+        pdfBlob = inBlob;
+        password = inPassword;
+        title = "";
+    }
+
+    /**
+     * Constructor with a DocumentModel. Uses the default <code>file:content</code> xpath to get the blob from the
+     * document.
+     *
+     * @param inDoc Input DocumentModel.
+     */
+    public PDFInfo(DocumentModel inDoc) {
+        this(inDoc, null, null);
+    }
+
+    /**
+     * Constructor for DocumentModel + encrypted PDF
+     * <p>
+     * If <inXPath</code> is <code>null</code> or "", it is set to the default
+     * <code>file:content</code> value.
+     *
+     * @param inDoc Input DocumentModel.
+     * @param inXPath Input XPath.
+     * @param inPassword If the PDF is encrypted.
+     */
+    public PDFInfo(DocumentModel inDoc, String inXPath, String inPassword) {
+        if (inXPath == null || inXPath.isEmpty()) {
+            inXPath = "file:content";
+        }
+        pdfBlob = (Blob) inDoc.getPropertyValue(inXPath);
+        password = inPassword;
+        title = "";
+    }
+
+    /**
+     * If set to true, parsing will extract PDF.
+     * <p>
+     * The value cannot be modified if <code>run()</code> already has been called.
+     *
+     * @param inValue true to extract XMP.
+     */
+    public void setParseWithXMP(boolean inValue) {
+        if (alreadyParsed && doXMP != inValue) {
+            throw new NuxeoException("Value of 'doXML' cannot be modified after the blob has been already parsed.");
+        }
+        doXMP = inValue;
+    }
+
+    private String checkNotNull(String inValue) {
+        return inValue == null ? "" : inValue;
+    }
+
+    /**
+     * After building the object with the correct constructor, and after possibly having set some parsing property
+     * (<code>setParseWithXMP()</code>, for example), this method will extract the information from the PDF.
+     * <p>
+     * After extraction, the info is available through getters: Either all of them (<code>toHashMap()</code> or
+     * <code>toString()</code>) or individual info (see all getters).
+     *
+     * @throws NuxeoException
+     */
+    public void run() throws NuxeoException {
+        // In case the caller calls several time the run() method
+        if (!alreadyParsed) {
+            fileName = pdfBlob.getFilename();
+            File pdfFile = pdfBlob.getFile();
+            if (pdfFile == null) {
+                fileSize = -1;
+            } else {
+                fileSize = pdfFile.length();
+            }
+            try {
```

Comment:
You can use try with resources since PDDocument is Closeable:
`try (PDDocument pdfDoc = PDDocument.load(pdfBlob.getStream())){`